src/lib/tpm.c

```diff
@@ -450,9 +450,6 @@ bool tpm_getrandom(tpm_ctx *ctx, BYTE *data, size_t size) {
 
     while (size) {
 
-        UINT16 requested_size = size > sizeof(rand_bytes->buffer) ?
```

Comment:
Hm, actually maybe we should keep this and pass it as the Esys_GetRandom argument?

Since the Esys_GetRandom size argument is only UINT16, requests for 65536 or more bytes will fail when they get truncated.

Also, if the TPM supports a larger TPM2B_DIGEST than the TSS does (unlikely, but theoretically possible), I think Esys_GetRandom will fail with TSS2_MU_RC_INSUFFICIENT_BUFFER.

Reply:
Done.